generators/pkg/generator/generate.go

```diff
@@ -25,3 +45,94 @@ func GenerateTask(spec GitHubSpec) v1beta1.Task {
 	)
 	return task
 }
+
+func addTask(url string) (v1beta1.Task, error) {
+	task := v1beta1.Task{}
+	response, err := http.Get(url)
+	if err != nil {
+		return task, fmt.Errorf("fail to get git-clone task from %s: %w", url, err)
+	}
+	defer response.Body.Close()
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return task, fmt.Errorf("fail to read reponse body: %v", err)
+	}
+
+	if err := yaml.Unmarshal(body, &task); err != nil {
+		return task, fmt.Errorf("unable to unmarshal git-clone task: %w", err)
+	}
+	task.Labels = map[string]string{"generator.tekton.dev": task.Name}
+	return task, nil
+}
+
+func GeneratePipeline(github GitHub) (v1beta1.Pipeline, []v1beta1.Task, error) {
```

Comment:
This func needs a comment. In particular, why are we returning both a Pipeline and a list of tasks?

Reply:
My idea is that when users want to `show` or `write` the generated config, the content will be the tasks used and a pipeline. This functions calls the `GenerateTask` and `addTask`. In this way I don't need to call these two functions again in the CLI to show the generated tasks. I am not sure if this strategy is good. 
The other one that I come up with is only return the pipeline here. But in the cli command, I need to call `GenerateTask` and `addTask` again.